### PR TITLE
feat: pyarrow `item`, `is_empty`, `rename`, `write_parquet`

### PR DESCRIPTION
**Pull request by FBruzzesi**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue #372 

## Checklist

- [x] Code follows style guide (ruff)
- [x] Tests added 
- [x] Documented the changes

## If you have comments or can explain your changes, please do so below.

Can this be considered addiction 😂🙈?

### CODE REVIEW

File: narwhals/_arrow/dataframe.py [hunk 1]
```diff
@@ -324,3 +329,55 @@ def collect(self) -> ArrowDataFrame:
...
+        df = self._native_dataframe
+
+        num_rows = df.num_rows
+        columns = df.column_names
+
+        for row in range(num_rows):
+            if not named:
+                yield tuple(df[_col][row].as_py() for _col in columns)
+            else:
+                yield {_col: df[_col][row].as_py() for _col in columns}
```
Comment (FBruzzesi) [inline narwhals/_arrow/dataframe.py:374]: that's a lot of `.as_py` calls, should we be concern about performance here?

File: narwhals/dependencies.py [hunk 2]
```diff
@@ -51,6 +51,15 @@ def get_pyarrow_compute() -> Any:  # pragma: no cover
     return None
 
 
+def get_pyarrow_parquet() -> Any:  # pragma: no cover
+    """Get pyarrow.parquet module (if pyarrow has already been imported - else return None)."""
+    if "pyarrow" in sys.modules:
+        import pyarrow.parquet as pp
+
+        return pp
+    return None
```
Comment (FBruzzesi) [inline narwhals/dependencies.py:60]: Not really necessary... thoughts?

File: narwhals/_arrow/dataframe.py (hunk 1 above)
Comment (MarcoGorelli) [reply narwhals/_arrow/dataframe.py:374]: hmm maybe...when I asked about iter_rows vs rows to polars devs, they said that doing it in buffers made a massive difference

if there's not a way to do that in pyarrow, maybe we can leave this one out for now?

File: narwhals/dependencies.py (hunk 2 above)
Comment (MarcoGorelli) [reply narwhals/dependencies.py:60]: it's fine 😄

File: narwhals/_arrow/dataframe.py (hunk 1 above)
Comment (FBruzzesi) [reply narwhals/_arrow/dataframe.py:374]: I was not able to find any reference on how to get an entire row performantly, especially as a python object.

> we can leave this one out for now?

What about adding a performance warning? After all, one should know that arrow has columnar data format, and we are asking for row iteration 🙈

Comment (FBruzzesi) [reply narwhals/_arrow/dataframe.py:374]: Ok I probably need to investigate further then. Let me leave `iter_rows` out of this PR

**Review by MarcoGorelli — APPROVED** [review]
thanks @FBruzzesi !

**Pull request merged**